mplc/partner.py

```diff
@@ -32,12 +31,39 @@ def __init__(self, partner_id):
         self.y_val = None
         self.y_test = None
 
-    def corrupt_labels(self):
-        for label in self.y_train:
+    def corrupt_labels(self, proportion_corrupted):
+        if not 0 < proportion_corrupted < 1:
+            raise ValueError(
+                f"The proportion of labels to corrupted  {proportion_corrupted} must be between 0 and 1."
+            )
+
+        # Select the indices where the label will be off-set
+        n = int(len(self.y_train) * proportion_corrupted)
+        idx = sample(list(range(len(self.y_train))), n)
+
+        # Off-set  the labels
+        for label in self.y_train[idx]:
             idx_max = np.argmax(label)
             label[idx_max] = 0.0
             label[idx_max - 1] = 1.0
```

Comment:
So here the hypothesis is that labels will always be `0` or `1`. @arthurPignet is that coherent / robust with regard to how you refactored the `Dataset` object and the way it is supposed to be used?